platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -19,6 +19,26 @@ const DEFAULT_VIEWPORT_OPTIONS: HangingProtocol.ViewportOptions = {
   viewportType: 'stack',
 };
 
+/**
+ * Returns an array of unique values for the given attribute from a series array.
+ * If the attribute is not present on the series, attempts to get it from the first instance.
+ * @param {Array} series - The series array to extract attributes from.
+ * @param {string} attribute - The attribute name to extract.
+ * @returns {Array} Array of unique attribute values.
+ */
+function getUniqueAttributeFromSeries(series, attribute) {
```

Comment:
Can you move this to the utilities, can call it getUniqueAttributeFromList
This is a generally useful function and it shouldn't stay embedded in the hanging protocol service.